### PR TITLE
openscap: architecture should'nt be empty

### PR DESCRIPTION
**Pull request by enoodle**

### CODE REVIEW

File: pkg/openscap/openscap.go [hunk 1]
```diff
@@ -109,11 +109,18 @@ func (s *defaultOSCAPScanner) getInputCVE(dist int) (string, error) {
...
+	}
+	return s
+}
+
 func (s *defaultOSCAPScanner) setOscapChrootEnv() error {
 	for k, v := range map[string]string{
 		"OSCAP_PROBE_ROOT":              s.imageMountPath,
 		"OSCAP_PROBE_OS_VERSION":        "4.2.6-301.fc23.x86_64", // FIXME place holder value
-		"OSCAP_PROBE_ARCHITECTURE":      s.image.Config.Labels["Architecture"],
+		"OSCAP_PROBE_ARCHITECTURE":      strOrDefault(s.image.Architecture, "NoArch"),
```
Comment (simon3z) [inline pkg/openscap/openscap.go]: @pweil- @enoodle are we sure `"NoArch"` is the appropriate default here? I am worried that OpenSCAP may not deal well with that (please work with @mpreisler on this one).

Comment (enoodle) [inline pkg/openscap/openscap.go]: @simon3z according to this: https://github.com/simon3z/image-inspector/pull/3#discussion_r58065097 it is not an important value. We just have to set it to something not empty other wise oscap fails.

Comment (mpreisler) [inline pkg/openscap/openscap.go]: OpenSCAP will accept any string but I think "Unknown" may be a better default value than "NoArch".

This value is saved in the ARF results and shown to user in the HTML report.

Comment (enoodle) [inline pkg/openscap/openscap.go]: @simon3z @mpreisler changed to "unknown"

File: pkg/openscap/openscap.go [hunk 2]
```diff
@@ -109,11 +109,18 @@ func (s *defaultOSCAPScanner) getInputCVE(dist int) (string, error) {
 	return cveFileName, err
 }
 
+func strOrDefault(s string, d string) string {
```
Comment (simon3z) [inline pkg/openscap/openscap.go]: @pweil- can you review this? Thanks.

**Comment by enoodle** [comment]
I removed this function and moved this logic to the ```Scan``` method. I also added logging there.

**Comment by pweil-** [comment]
:+1:

File: pkg/openscap/openscap.go [hunk 3]
```diff
@@ -158,6 +161,12 @@ func (s *defaultOSCAPScanner) Scan(mountPath string, image *docker.Image) error
 	s.image = image
 	s.imageMountPath = mountPath
 
+	s.imageArch = image.Architecture
+	if len(s.imageArch) == 0 {
+		log.Printf("Unable to identify image's architecture")
```
Comment (simon3z) [inline pkg/openscap/openscap.go]: @enoodle I apologize, I forgot this is in the pkg library. Please remove this line and the `"log"` import above. Other than that :+1:

File: pkg/openscap/openscap.go [hunk 4]
```diff
@@ -158,6 +161,12 @@ func (s *defaultOSCAPScanner) Scan(mountPath string, image *docker.Image) error
...
+		s.imageArch = "Unknown"
```
Comment (simon3z) [inline pkg/openscap/openscap.go]: @enoodle since you're at it maybe we can make a constant out of `Unknown`

**Comment by enoodle** [comment]
@simon3z I went back to the original implementation and added the constant.

**Comment by simon3z** [comment]
@enoodle do we have tests for this?

**Comment by enoodle** [comment]
@simon3z I added tests for ```strOrDefault``` is this what you wanted? (If so I will squash the commits)

**Comment by simon3z** [comment]
@enoodle I would have preferred to check more extensively that the OpenSCAP env variables are set correctly for a various number of cases.

File: pkg/openscap/openscap.go [hunk 5]
```diff
@@ -109,11 +110,18 @@ func (s *defaultOSCAPScanner) getInputCVE(dist int) (string, error) {
...
+	if len(s) == 0 { // s || d
+		return d
+	}
+	return s
+}
+
 func (s *defaultOSCAPScanner) setOscapChrootEnv() error {
 	for k, v := range map[string]string{
 		"OSCAP_PROBE_ROOT":              s.imageMountPath,
 		"OSCAP_PROBE_OS_VERSION":        "4.2.6-301.fc23.x86_64", // FIXME place holder value
```
Comment (simon3z) [inline pkg/openscap/openscap.go]: @enoodle what's this place holder??? please fix with something more appropriate.

File: pkg/openscap/openscap.go (hunk 2 above)
Comment (pweil-) [inline pkg/openscap/openscap.go]: looks fine to me, I really wish there was a ternary operator in golang for things like this (never thought I'd be wishing for that).  Only suggestion is to move to a util package and make it public so we don't rewrite it again.

Comment (simon3z) [inline pkg/openscap/openscap.go]: :+1: yeah moving it to util makes sense.

**Comment by enoodle** [comment]
added the ```util``` package. I will add more tests later today/ tomorrow morning

**Comment by enoodle** [comment]
@simon3z @pweil- Tests were added. Please review.

**Comment by simon3z** [comment]
@enoodle can `Architecture` be `nil`? (Maybe in old Docker versions...) Maybe it's better to protect against that as well.

**Comment by enoodle** [comment]
@simon3z its of type ```string``` not a pointer so it can't be ```nil```.

**Pull request merged**